packages/loader/driver-utils/src/runWithRetry.ts

```diff
@@ -48,6 +48,7 @@ export async function runWithRetry<T>(
 	fetchCallName: string,
 	logger: ITelemetryLoggerExt,
 	progress: IProgress,
+	beforeErrorThrow?: () => void,
```

Comment:
this shouldn't be necessary, just do the same thing in a .catch of the promise returned by the function